api.planx.uk/inviteToPay/paymentRequest.ts

```diff
@@ -0,0 +1,112 @@
+import { gql } from "graphql-request";
+import { NextFunction, Request, Response } from "express";
+import { adminGraphQLClient as client } from "../hasura";
+import { ServerError } from "../errors";
+import {
+  postPaymentNotificationToSlack,
+  fetchPaymentViaProxyWithCallback,
+} from "../pay";
+import type { GovUKPayment } from "../types";
+
+// middleware used by routes:
+//  * /payment-request/:paymentRequest/pay
+//  * /payment-request/:paymentRequest/payment/:paymentId
+export async function fetchPaymentRequestDetails(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const query = gql`
+    query GetPaymentRequestByID($paymentRequestId: uuid!) {
+      payment_requests_by_pk(id: $paymentRequestId) {
+        session_id
+        payment_amount
+        session {
+          flow_id
+          flow {
+            team {
+              slug
+            }
+          }
+        }
+      }
+    }
+  `;
+  const { payment_requests_by_pk } = await client.request(query, {
+    paymentRequestId: req.params.paymentRequest,
+  });
+  if (!payment_requests_by_pk) {
+    return next(
+      new ServerError({
+        message: "payment request not found",
+        status: 404,
+      })
+    );
+  }
+  const sessionId = payment_requests_by_pk.session_id;
+  if (sessionId) req.query.sessionId = sessionId;
+
+  const localAuthority = payment_requests_by_pk.session?.flow?.team?.slug;
+  if (localAuthority) req.params.localAuthority = localAuthority;
+
+  const flowId = payment_requests_by_pk.session?.flow_id;
+  if (flowId) req.query.flowId = flowId;
+
+  const paymentAmount = payment_requests_by_pk.payment_amount;
+  if (paymentAmount) req.params.paymentAmount = paymentAmount;
+
+  next();
+}
```

Comment:
This middleware sets up the request to be compatible with the pay proxy middleware.